### PR TITLE
Part 2 of 3090: add glossary term approval workflow

### LINKED ISSUE #3090: Add support for Glossary Term approval workflow
**Feature**
#2517 

## Glossary Term Approval workflow
This feature adds support for the Glossary Term Approval workflow. This functions as follows:
1. If a Glossary has `Reviewers` set, any glossary term created in it will be `Draft` mode. Only `Reviewers` can change the `Draft` status to `Approved` or `Rejected`. Event the owners of the Glossary can't approve or reject the newly added term.
2. If a Glossary does **not** have `Reviewers` set, any glossary term created under it is created in `Approved` mode.

## Tasks:
- [x] Part1 - Add backend support for glossary workflows
- [ ] Part 2 - Add support for approval workflow in the UI - cc @chirag-madlani 
- [ ] Part 3 - Only show `Approved` glossary term for labeling - cc @harshach

### PR DESCRIPTION
**Pull request by karanh37**
### Describe your changes:

Fixes #3090 Part 2

Add Glossary Term Approval Workflow
Hide Glossary Left panel for glossary terms page
Show status badge in header

I worked on ... because ...

https://github.com/open-metadata/OpenMetadata/assets/33024356/19374805-00fd-4fb0-a5e7-e03c8a156ec6

https://github.com/open-metadata/OpenMetadata/assets/33024356/f15afbc7-d3a3-4ab7-ba0e-1796bd6dd3d4

Screenshot 2023-09-25 at 10 33 59 PM

#
### Type of change:

- [x] Improvement

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/developers/contribute) document.
- [ ] My PR title is `Fixes <issue-number>: <short explanation>`
- [ ] I have commented on my code, particularly in hard-to-understand areas. 
- [ ] For JSON Schema changes: I updated the migration scripts or explained why it is not needed.

### CODE REVIEW

File: openmetadata-ui/src/main/resources/ui/src/components/Task/TaskTab/TaskTab.component.tsx [hunk 1]
```diff
@@ -310,6 +345,7 @@ export const TaskTab = ({
     handleMenuItemClick,
     taskAction,
     isTaskClosed,
+    isTaskGlossaryApproval,
```
Comment (chirag-madlani) [inline openmetadata-ui/src/main/resources/ui/src/components/Task/TaskTab/TaskTab.component.tsx:348]:
```suggestion
    isTaskGlossaryApproval,
    approvalWorkflowActions,
```

File: openmetadata-ui/src/main/resources/ui/cypress/e2e/Pages/Glossary.spec.js [hunk 2]
```diff
@@ -402,28 +419,44 @@ const voteGlossary = (isGlossary) => {
   initialVoting(isGlossary ? '@voteGlossary' : '@voteGlossaryTerm');
 };
 
+const goToGlossaryPage = () => {
+  interceptURL('GET', '/api/v1/glossaryTerms*', 'getGlossaryTerms');
+  interceptURL('GET', '/api/v1/glossaries?fields=*', 'fetchGlossaries');
+  cy.get('[data-testid="governance"]')
+    .should('exist')
+    .and('be.visible')
```
Comment (chirag-madlani) [inline openmetadata-ui/src/main/resources/ui/cypress/e2e/Pages/Glossary.spec.js:427]: can remove this

File: openmetadata-ui/src/main/resources/ui/cypress/e2e/Pages/Glossary.spec.js [hunk 3]
```diff
@@ -402,28 +419,44 @@ const voteGlossary = (isGlossary) => {
...
+    .click({ animationDistanceThreshold: 20 });
+
+  // Clicking on Glossary
+  cy.get('.govern-menu')
+    .should('exist')
+    .and('be.visible')
```
Comment (chirag-madlani) [inline openmetadata-ui/src/main/resources/ui/cypress/e2e/Pages/Glossary.spec.js:433]: can remove this

File: openmetadata-ui/src/main/resources/ui/cypress/e2e/Pages/Glossary.spec.js [hunk 4]
```diff
@@ -402,28 +419,44 @@ const voteGlossary = (isGlossary) => {
...
+    .then(($el) => {
+      cy.wrap($el)
+        .find('[data-testid="app-bar-item-glossary"]')
+        .should('exist')
+        .and('be.visible')
```
Comment (chirag-madlani) [inline openmetadata-ui/src/main/resources/ui/cypress/e2e/Pages/Glossary.spec.js:438]: can remove this

File: openmetadata-ui/src/main/resources/ui/src/components/ActivityFeed/ActivityFeedTab/ActivityFeedTab.component.tsx [hunk 5]
```diff
@@ -111,7 +111,13 @@ export const ActivityFeedTab = ({
     history.push(
       getEntityDetailLink(
         entityType,
-        EntityType.TABLE === entityType ? getEncodedFqn(fqn) : fqn,
+        [
+          EntityType.TABLE,
+          EntityType.GLOSSARY,
+          EntityType.GLOSSARY_TERM,
+        ].includes(entityType)
+          ? getEncodedFqn(fqn)
```
Comment (chirag-madlani) [inline openmetadata-ui/src/main/resources/ui/src/components/ActivityFeed/ActivityFeedTab/ActivityFeedTab.component.tsx:119]: Is there a way we could avoid this condition

File: openmetadata-ui/src/main/resources/ui/src/components/Glossary/GlossaryTermTab/GlossaryTermTab.component.tsx [hunk 6]
```diff
@@ -123,6 +125,24 @@ const GlossaryTermTab = ({
         key: 'owner',
         render: (owner: EntityReference) => <OwnerLabel owner={owner} />,
       },
+      {
+        title: t('label.status'),
+        dataIndex: 'status',
+        key: 'status',
+        filters: StatusFilters,
```
Comment (chirag-madlani) [inline openmetadata-ui/src/main/resources/ui/src/components/Glossary/GlossaryTermTab/GlossaryTermTab.component.tsx:132]: `filterIcon` is missing here, we should have that

File: openmetadata-ui/src/main/resources/ui/src/pages/Glossary/GlossaryPage/GlossaryPage.component.tsx [hunk 7]
```diff
@@ -180,12 +168,6 @@ const GlossaryPage = () => {
     }
   }, [isGlossaryActive, glossaryFqn, glossaries]);
 
-  useEffect(() => {
-    if (!isGlossaryActive && viewAllGlossaryPermission) {
-      fetchGlossaryTermParent();
-    }
-  }, [glossaryFqn]);
```
Comment (chirag-madlani) [inline openmetadata-ui/src/main/resources/ui/src/pages/Glossary/GlossaryPage/GlossaryPage.component.tsx:187]: Why did we remove this? We don't need it now?

Comment (karanh37) [reply openmetadata-ui/src/main/resources/ui/src/pages/Glossary/GlossaryPage/GlossaryPage.component.tsx:187]: This was added during permissions thing which is making duplicate API calls.
Not needed.

File: openmetadata-ui/src/main/resources/ui/cypress/common/common.js [hunk 8]
```diff
@@ -1301,3 +1302,79 @@ export const visitDataModelPage = (dataModelFQN, dataModelName) => {
 
   verifyResponseStatusCode('@getDataModelDetails', 200);
 };
+
+export const signupAndLogin = (email, password, firstName, lastName) => {
+  return new Cypress.Promise((resolve) => {
```
[the remark this answers is not in the document]
Comment (karanh37) [reply openmetadata-ui/src/main/resources/ui/cypress/common/common.js:1307]: When we create a user, we needed ID of the created user, which can then be used to make delete call.
So created this Promise

**Review by chirag-madlani — APPROVED** [review]

**Review by harshach — APPROVED** [review]

**Pull request merged**